pkg/resource/deployment.go

```diff
@@ -95,8 +96,17 @@ func (gd *generatorDeployment) expected() (runtime.Object, error) {
 
 	// Strategy defaults to RollingUpdate
 	deployStrategy := gd.cr.Spec.RolloutStrategy
+	var rollingUpdate *appsapi.RollingUpdateDeployment
 	if deployStrategy == "" {
 		deployStrategy = string(appsapi.RollingUpdateDeploymentStrategyType)
+		if gd.cr.Spec.Replicas == 2 {
```

Comment:
Do you want to do it only when `rolloutStrategy ` is not set?

Reply:
That was my initial thought but yeah, I guess we can always set it. Changing it, PTAL.